samples/diagnostics/dumping/Dumping States and Operations.ipynb

```diff
@@ -0,0 +1,1035 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/json": "{\"LastUpdated\":\"2020-08-26T09:39:52.2294341-07:00\",\"IsCompleted\":true,\"Description\":\"Adding package Microsoft.Quantum.Standard.Visualization\",\"Subtask\":\"done\"}",
+      "text/plain": [
+       "Adding package Microsoft.Quantum.Standard.Visualization: done!"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "application/json": "[\"Microsoft.Quantum.Standard::0.12.20082513\",\"Microsoft.Quantum.Standard.Visualization::0.12.20082513\"]",
+      "text/html": [
+       "<ul><li>Microsoft.Quantum.Standard::0.12.20082513</li><li>Microsoft.Quantum.Standard.Visualization::0.12.20082513</li></ul>"
+      ],
+      "text/plain": [
+       "Microsoft.Quantum.Standard::0.12.20082513, Microsoft.Quantum.Standard.Visualization::0.12.20082513"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%package Microsoft.Quantum.Standard.Visualization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For convinence, we'll also open the [Microsoft.Quantum.Diagnostics namespace](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.diagnostics) so that its functions and operations are available throughout the notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/json": "[]",
+      "text/html": [
+       "<ul></ul>"
+      ],
+      "text/plain": []
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Diagnostics;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dumping Simulator States"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When using Q# programs with a simulator, it can be helpful to ask the simulator for additional diagnostics that aren't available when running on quantum hardware. In particular, when using the `QuantumSimulator` as your target (e.g.: with the `%simulate` command), calling [`DumpMachine`](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.diagnostics.dumpmachine) and [`DumpRegister`](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.diagnostics.dumpregister) will show you the state internal to the simulator.\n",
+    "\n",
+    "Since `DumpMachine` returns `Unit` and has no effect on the state of any qubits, it can be safely replaced by a no-op on targets that don't support any additional diagnostics, such as quantum hardware."
```

Comment:
This sounds a bit like as the user has to actively do something to replace `DumpMachine` by a no-op. I suggest to rephrase it that simulators handle `DumpMachine` as no-op when targeting hardware.

Reply:
Good catch here as well, thanks.